src/suspense.js

```diff
@@ -0,0 +1,68 @@
+import { Component } from './component';
+import { createElement } from './create-element';
+
+// TODO: react warns in dev mode about defaultProps and propTypes not being supported on lazy
+// loaded components
+
+export const sym = Symbol.for('Suspense');
+
+export class Suspense extends Component {
+	constructor(props) {
+		// TODO: should we add propTypes in DEV mode?
+		super(props);
+
+		// mark this component as a handler of suspension (thrown Promises)
+		this[sym] = sym;
+
+		this.state = {
+			l: false
+		};
+	}
+
+	componentDidCatch(e) {
+		if (e && typeof e.then === 'function') {
+			this.setState({ l: true });
+			e.then(
+				() => {
+					this.setState({ l: false });
+				},
+				// Suspense ignores errors thrown in Promises as this should be handled by user land code
+				() => {
+					this.setState({ l: false });
+				}
+			);
+		}
+		else {
+			throw e;
+		}
+	}
+
+	render() {
+		return this.state.l ? this.props.fallback : this.props.children;
```

Comment:
Using the method arguments (only available in Preact) saves another `6 B`.

```js
render(props, state) {
  return state.l ? props.fallback : props.children;
}
```

Reply:
Done